extras/src/main/java/com/google/gson/graph/GraphAdapterBuilder.java

```diff
@@ -98,10 +98,10 @@ public void registerOn(GsonBuilder gsonBuilder) {
             out.nullValue();
             return;
           }
+          writeTypeAdapter( out, value);
+        }
 
-          Graph graph = graphThreadLocal.get();
-          boolean writeEntireGraph = false;
-
+        public void writeTypeAdapter(JsonWriter out, T value) throws IOException {
```

Comment:
Not sure if these few lines really have any effect on the complexity (same for the `readTypeAdapter` method below). Maybe this change reduces a theoretical complexity value; but in reality I would say it does not make a (big) difference.

Reply:
I typed up a longer comment which seems to have got lost. I was suggesting pulling original lines 122-124 into a method and getting rid of the `writeEntireGraph` variable. I think that would be a better way of simplifying the code than just separating out the null case.